File: src/main/java/com/lessmarkup/engine/data/ConnectionManager.java
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */
package com.lessmarkup.engine.data;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public final class ConnectionManager {
    public static Connection getConnection(String connectionString) throws SQLException, ClassNotFoundException {
        
        int driverPart = connectionString.indexOf(';');
        if (driverPart > 0 && driverPart < connectionString.indexOf(':')) {
            String driverClass = connectionString.substring(0, driverPart);
            connectionString = connectionString.substring(driverPart+1);
            Class.forName(driverClass);
        }
        
        return DriverManager.getConnection(connectionString);
    }
}

File: src/main/scala/com/lessmarkup/engine/data/ConnectionManager.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.engine.data

import java.sql.{Connection, DriverManager}

object ConnectionManager {
  def getConnection(connectionString: String): Connection = {
    val driverPart: Int = connectionString.indexOf(';')
    val connectionStringParsed = if (driverPart > 0 && driverPart < connectionString.indexOf(':')) {
      val driverClass: String = connectionString.substring(0, driverPart)
      Class.forName(driverClass)
      connectionString.substring(driverPart + 1)
    } else {
      connectionString
    }
    DriverManager.getConnection(connectionStringParsed)
  }
}